singleton.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package zap
+
+import "sync"
+
+var (
+	singletonLogger *logger
+	initOnce        sync.Once
+	configLock      sync.Mutex
+)
+
+// Standard returns the standard, singleton logger. Although this behavior is discouraged in many settings
+// many projects use it regardless and we'd prefer a reference implementation to each service doing it differently.
+func Standard() Logger {
+	if singletonLogger == nil {
```

Comment:
This can race with `ConfigureStandard`. Currently `configLock` protects against calling `ConfigureStandard` in parallel, but there's still a race if someone calls it while trying to read it.

I don't really think we want a lock when trying to use the logger though. My preference is to instead not have any synchronization, and make it clear that `ConfigureStandard` cannot be called in parallel with `Standard`, and initialize it in the declaration, e.g.,
```go
var singletonLogger = ConfigureStandard(NewJSONEncoder(...))
```

We can then remove the lock and the `sync.Once`